tests/gen/gen_auto_config.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package gen
+
+import (
+	"reflect"
+
+	"github.com/sourcenetwork/defradb/client"
+)
+
+// genConfig is a configuration for a generation of a field.
+type genConfig struct {
+	labels         []string
+	props          map[string]any
+	fieldGenerator GenerateFieldFunc
+}
+
+// configsMap is a map of type name to a map of field name to a generation configuration.
+type configsMap map[string]map[string]genConfig
```

Comment:
praise: Thanks for aliasing and doucmenting this, it makes the referencing code much easier to read